packages/backend-storage/src/factory.ts

```diff
@@ -46,10 +49,68 @@ class AmplifyStorageFactory
         getInstanceProps
       );
     }
-    return constructContainer.getOrCompute(this.generator) as AmplifyStorage;
+    const amplifyStorage = constructContainer.getOrCompute(
+      this.generator
+    ) as AmplifyStorage;
+    Aspects.of(amplifyStorage).add(
+      new StorageValidator(Stack.of(amplifyStorage))
```

Comment:
Instead of creating an aspect for the individual `AmplifyStorage`, you can just create one for the nested stack which will get executed just once and you wouldn't have to pass the nested stack in the constructor

```suggestion
    Aspects.of(Stack.of(amplifyStorage)).add(
      new StorageValidator()
```